plugin.py

```diff
@@ -29,99 +28,46 @@ def is_allowed_to_start(
         initiating_view: Optional[sublime.View] = None,
         workspace_folders: Optional[List[WorkspaceFolder]] = None,
         configuration: Optional[ClientConfig] = None
-    ):
+    ) -> Optional[str]:
         if not workspace_folders or not configuration:
             return
-        configuration.init_options.set('textDocumentSync', get_text_document_sync(configuration))
-        configuration.init_options.set('languageFeatures', get_language_features(configuration))
-        configuration.init_options.set('documentFeatures', {
-            "selectionRange": True,
-            "foldingRange": True,
-            "linkedEditingRange": False,
-            "documentSymbol": True,
-            "documentColor": True,
-            "documentFormatting": {
-                "defaultPrintWidth": 90
-            }
-        })
-        if configuration.init_options.get('typescript.serverPath'):
-            return  # don't find the `typescript.serverPath` if it was set explicitly in LSP-volar.sublime-settings
-        typescript_path = cls.find_typescript_path(workspace_folders[0].path)
-        configuration.init_options.set('typescript.serverPath', typescript_path)
+        if configuration.init_options.get('typescript.tsdk'):
+            return  # don't find the `typescript.tsdk` if it was set explicitly in LSP-volar.sublime-settings
+        typescript_lib_path = cls.find_typescript_lib_path(workspace_folders[0].path)
+        if not typescript_lib_path:
+            return 'Could not resolve location of TypeScript package'
+        configuration.init_options.set('typescript.tsdk', typescript_lib_path)
 
     @classmethod
-    def find_typescript_path(cls, current_folder: str) -> str:
+    def find_typescript_lib_path(cls, workspace_folder: str) -> Optional[str]:
```

Comment:
this logic is not backwords compatible with the old code:
```ts
    const parent = path.resolve(dir, '..');
    if (parent !== dir) {
        return findPathToModule(parent, MODULE_PATHS);
    }
```
Previous logic also scanned up parent directories until it got to the root directory.

Reply:
Yes, i know, but it seemed weird to me to look beyond the workspace folder.

Reply:
I tried to remember why we did it like that, 
it was copy pasted from // https://github.com/typescript-language-server/typescript-language-server/blob/6fe20dac1c28c684cf3214bfdbd98f05225090b9/src/utils/modules-resolver.ts 

I am ok with removing the look up in the parent dirs.
If it causes some issues, we will resolve it.

Reply:
Thing to consider though is that tls allows single files without workspace in which it makes more sense to check higher directories. In our case it seems like we don't really allow that. (Though at the same time we don't prevent volar from starting which seems like a bug).

Reply:
I've tried volar with no workspace and it falling back to bundled TS and while it doesn't crash, it doesn't really seem to provide any functionality either so I think we should prevent it from starting if there is no worspace.